update some hashes, and update maintainers list added erkin (me)

erkingoksuu opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other contributing documentation in corresponding paths.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Tibso committed:
maintainers: add Tibso

Tibso committed:
python3Packages.amulet-nbt: init at 2.1.5

Co-authored-by: Leah Amelia Chen <hi@pluie.me>
Co-authored-by: langsjo <104687438+langsjo@users.noreply.github.com>

Tibso committed:
python3Packages.amulet-leveldb: init at 1.0.2

Co-authored-by: Leah Amelia Chen <hi@pluie.me>

Tibso committed:
python3Packages.pymctranslate: init at 1.2.33

Co-authored-by: Leah Amelia Chen <hi@pluie.me>

Tibso committed:
python3Packages.amulet-core: init at 1.9.29

Co-authored-by: Leah Amelia Chen <hi@pluie.me>

Tibso committed:
python3Packages.minecraft-resource-pack: init at 1.4.6

Co-authored-by: Leah Amelia Chen <hi@pluie.me>

Tibso committed:
amulet-map-editor: init at 0.10.42

Co-authored-by: Leah Amelia Chen <hi@pluie.me>

Tibso committed:
lib/licenses: add amulet

Co-authored-by: Leah Amelia Chen <hi@pluie.me>

erkingoksuu committed:
amulet-map-editor: init 0.10.42 -> init 0.10.44, fix build and runtime

This commit updates Amulet Map Editor to version 0.10.44 and applies
several critical fixes to make the package buildable and stable on
NixOS.

Key changes include:

1- Runtime Crash Fix: Resolved a `GLib-GIO-ERROR` that occurred when
  opening the file picker. The application is now correctly wrapped using
  `wrapGAppsHook3` to provide the necessary GSettings schemas and desktop
  integration.

    In detail:
      fix: Wrap package to resolve file picker crash

      The `amulet-map-editor` application would consistently crash with a
      `GLib-GIO-ERROR` upon attempting to open a file dialog. The root
      cause was the absence of GSettings schemas in the pure Nix environment,
      a common issue for applications with a GTK backend (used by wxPython
      on Linux).

      The issue is fixed by vendoring the package derivation and
      correctly preparing its runtime environment.

      i.  Wrap Application: The derivation is now processed by the
          `wrapGAppsHook3`. This hook ensures that environment variables like
          `XDG_DATA_DIRS` are correctly populated at runtime, allowing the
          application to find its schemas and other resources.

      ii.  Add Runtime Dependencies: The following dependencies were added
          to the package's `buildInputs` to provide necessary runtime data:
          - `gsettings-desktop-schemas`: Provides the core desktop schemas.
          - `adwaita-icon-theme`: Provides a default icon set.
          - `xdg-desktop-portal-gtk`: Provides the GTK backend for the XDG
            file chooser portal, which is essential on Wayland.

2- Build Fix: Pinned the package to `python312` to resolve a build
  failure caused by a `numpy-1.26.4` with `python313` incompatibility.

3- Maintenance:
  - Updated hashes for several dependencies to ensure correct builds.
  - Added a new maintainer erkin (me) for the package.
  - Format fixed from,

  `maintainers = with lib.maintainers; [ tibso erkin ];`

   to,

   `maintainers = with lib.maintainers; [
      tibso
      erkin
    ];`

    for fix the format error.

alyssais reviewed (changes requested):
This looks like it needs a rebase, and also to be changed to follow the commit message conventions.

alyssais commented:
Commit messages still need fixed. What's this PR even trying to accomplish? There's no explanation.

erkingoksuu commented:
> Commit messages still need fixed. What's this PR even trying to accomplish? There's no explanation.

Thank you for your review.
I rebased and updated the all commits with squash it.

Explanation:
This pull request adds Amulet Map Editor and all of its required Python dependencies to Nixpkgs.

The initial packaging effort was started with contributions from @Tibso, @pluiedev, and @langsjo. Building on this foundation, I've added the following critical fixes and improvements to ensure the package runs smoothly and stably on NixOS:

    Runtime Crash Fix: The GTK/GSettings error that caused the application to crash when opening the file picker has been resolved by wrapping the package with wrapGAppsHook3. This ensures the application integrates correctly with the desktop environment.

    Dependency Issue Resolution: To fix the build failure caused by a numpy version incompatibility, the package has been pinned to use python312.

    Version Update and Maintenance: The package has been updated to the latest version (0.10.44), dependency hashes have been corrected, and the code style has been standardized using nixpkgs-fmt.

As a result of these changes, amulet-map-editor now works as expected on NixOS.

erkingoksuu reviewed (commented):
> Commit messages still need fixed. What's this PR even trying to accomplish? There's no explanation.

Thank you for your review.
I rebased and updated the all commits with squash it.

Explanation:
This pull request adds Amulet Map Editor and all of its required Python dependencies to Nixpkgs.

The initial packaging effort was started with contributions from @Tibso, @pluiedev, and @langsjo. Building on this foundation, I've added the following critical fixes and improvements to ensure the package runs smoothly and stably on NixOS:

    Runtime Crash Fix: The GTK/GSettings error that caused the application to crash when opening the file picker has been resolved by wrapping the package with wrapGAppsHook3. This ensures the application integrates correctly with the desktop environment.

    Dependency Issue Resolution: To fix the build failure caused by a numpy version incompatibility, the package has been pinned to use python312.

    Version Update and Maintenance: The package has been updated to the latest version (0.10.44), dependency hashes have been corrected, and the code style has been standardized using nixpkgs-fmt.

As a result of these changes, amulet-map-editor now works as expected on NixOS.

alyssais commented:
I'm sorry but I have to ask: am I talking to an LLM?

erkingoksuu commented:
I am sorry for that. But, since I couldn't explain the changes that can be easily understood from the mentioned #201016 request and the briefly mentioned commit messages, or maybe you can't understand it, I chose this way in case the LLMs can explain it better than me, or maybe you understand better when the LLM wrote. I think there is no need to be rude. Please make your review. Thanks.

erkingoksuu commented:
Thank you for your feedback. First of all, I would like to say that I do not mean to be rude and I apologize for the misunderstanding. In addition, I would like you to share the errors you find so that I can fix them. You provided “Applied nixpkgs-fmt for code style consistency.” as an example. I'll correct it immediately and explain it in more detail. Again, I'm sorry for the misunderstanding. I will be waiting for other errors you want me to fix as a review. Thank you.

pluiedev commented:
How is this different from #405545? I find it very problematic that you just decided to open a new PR that aims to replace the old one, with the commits from both it and my own PR, **without notifying either of us**. This isn't fair at all.

If you really want to propose changes, just use the PR review feature or send your patches as a link or as a gist or something. This as it stands is not collaboration.

pluiedev commented:
As for #405548, at least @Tibso pinged me at the outset and let me know they're reviving a stale PR — that to me was totally okay since I was properly notified, and they were willing to build upon what I wrote and credit me appropriately. I'm not convinced it's the same case here — #405548 isn't inactive and there doesn't *need* to be another successor PR. Therefore I find the whole intent behind this PR highly suspect and disingenuous at best.

Please, if you have any substantial changes, submit them to #405548. This is not acceptable.

pluiedev commented:
> I'm sorry but I have to ask: am I talking to an LLM?

Just as an aside, @alyssais I don't really think we should be immediately accusing people of using LLMs based on behavior like this. 

It's totally understandable, since I frankly don't know why anyone working in good faith would choose to silently open a PR like this instead of, y'know, collaborating on the original PR, but I think it might come off as a bit too aggressive against people who might genuinely be new to online collaboration, or don't speak English fluently enough to interact in this community without some kind of AI to help them communicate. 

I know it's somewhat of a kneejerk reaction, but let's try to be charitable towards others here — I certainly overreacted too myself I think, but the whole premise was just honestly silly.

pluiedev commented:
I guess this serves as an example for #410147 as an AI-generated PR then? It's a bit disheartening to see but I guess the doubters have to face that this is in fact what is happening now.

erkingoksuu commented:
Thanks for the feedback, and sorry if my intention wasn't clear.

Of course, before opening this PR, I looked at other PRs, and then I decided to open a new PR because this PR #405548 had no comments or reviews for a month, even though the tests showed no problems. In this context, assuming that this PR #405548 is now inactive, I opened a new active PR, including the commits that @Tibso, @pluiedev and @langsjo had already made.

@pluiedev accused me of theft. First of all, I find this very disrespectful. As I said, if I wanted to do something like this, I would have done it by copying it directly, right, I would not have added to the already existing commits. I would also like you to know that I congratulate you for your point of view because you called politeness as manipulation in your comment on issue #410741. I would be grateful if you could let me know since when politeness became manipulative. I expect you to apologize.

In addition, I think it would do us personally and the community a lot of good if we dynamically adapt to the direction the world is going and use LLMs in a positive way, as in this case.

Finally, as you indicated, I'll put my changes directly in #405548, thank you.

emilazy commented:
I apologize; I missed https://github.com/NixOS/nixpkgs/issues/410741#issuecomment-3041327084 and was going based on the comments in this PR. I agree that the accusation there is untrue and inappropriate.
